OpenRA.Mods.RA2/Activities/EnterGarrison.cs

```diff
@@ -0,0 +1,214 @@
+
+        public EnterGarrison(Actor self, Target garrisonableBuilding)
+        {
+            // Base - Enter Properties
+            move = self.Trait<IMove>();
+            this.target = garrisonableBuilding;
+
+            // EnterGarrison Properties
+            this.targetLineColor = null;
+            this.garrisonableBuilding = garrisonableBuilding;
+            garrisoner = self.TraitsImplementing<Garrisoner>().Single();
+        }
+
+        protected bool TryStartEnter(Actor self, Actor targetActor)
+        {
+            enterActor = targetActor;
+            enterGarrison = targetActor.TraitOrDefault<Garrisonable>();
+
+            // Make sure we can still enter the transport
+            // (but not before, because this may stop the actor in the middle of nowhere)
+            if (enterGarrison == null || !garrisoner.Reserve(self, enterGarrison))
+            {
+                Cancel(self, true);
+                return false;
+            }
+
+            return true;
+        }
+
+        protected void OnCancel(Actor self) { }
+
+        protected void OnEnterComplete(Actor self, Actor targetActor)
+        {
+            self.World.AddFrameEndTask(w =>
+            {
+                // Make sure the target hasn't changed while entering
+                // OnEnterComplete is only called if targetActor is alive
+                if (targetActor != enterActor)
+                    return;
+
+                if (!enterGarrison.CanLoad(enterActor, self))
+                    return;
+
+                enterGarrison.Load(enterActor, self);
+                w.Remove(self);
+
+                // Preemptively cancel any activities to avoid an edge-case where successively queued
+                // EnterTransports corrupt the actor state. Activities are cancelled again on unload
+                self.CancelActivity();
+            });
+        }
+
+        /// Base Enter Methods Below
+        public override Activity Tick(Actor self)
+        {
+            // Update our view of the target
+            bool targetIsHiddenActor;
+            target = target.Recalculate(self.Owner, out targetIsHiddenActor);
+
+            // Re-acquire the target after change owner has happened.
+            if (target.Type == TargetType.Invalid)
+                target = Target.FromActor(target.Actor);
+
+            if (!targetIsHiddenActor && target.Type == TargetType.Actor)
+                lastVisibleTarget = Target.FromTargetPositions(target);
+
+            var oldUseLastVisibleTarget = useLastVisibleTarget;
+            useLastVisibleTarget = targetIsHiddenActor || !target.IsValidFor(self);
+
+            // Cancel immediately if the target died while we were entering it
+            if (!IsCanceled && useLastVisibleTarget && lastState == EnterState.Entering)
+                Cancel(self, true);
+
+            // TickInner(self, target, useLastVisibleTarget); // Left Over from Enter. unused in EnterGarrison

```

Comment:
This rings a mior warning bell in my head. Should be checked if it causes any issues - if not, please remove as dead code.

Reply:
There are many other classes in OpenRA.Mods.* that Inherit from the Enter class and the only one I've found that implements TickInner is infiltrate from OpenRA.Mods.cnc. 

The correct fix for this I think would be to add a virtual tick inner function to EnterGarrison, as there is one in Enter, and just do nothing with it (the exact way it's done in Enter)

Reply:
If only Infiltrate implements it, then there is no need for this line. Remove.